samples/net/ipv4_autoconf/src/main.c

```diff
@@ -0,0 +1,65 @@
+/* Networking DHCPv4 client */
+
+/*
+ * Copyright (c) 2017 ARM Ltd.
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if 1
+#define SYS_LOG_DOMAIN "ipv4ll"
+#define NET_SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <zephyr.h>
+#include <linker/sections.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <net/net_if.h>
+#include <net/net_core.h>
+#include <net/net_context.h>
+#include <net/net_mgmt.h>
+
+static struct net_mgmt_event_callback mgmt_cb;
+
+static void handler(struct net_mgmt_event_callback *cb,
+		    u32_t mgmt_event,
+		    struct net_if *iface)
+{
+	int i = 0;
+
+	if (mgmt_event != NET_EVENT_IPV4_ADDR_ADD) {
+		return;
+	}
+
+	for (i = 0; i < NET_IF_MAX_IPV4_ADDR; i++) {
+		char buf[NET_IPV4_ADDR_LEN];
+
+		if (iface->config.ip.ipv4->unicast[i].addr_type != NET_ADDR_AUTOCONF) {
```

Comment:
Please search and replace all "iface->config" by a call to "net_if_get_config(iface)" so instead of iface->config.ip.ipv4... we would have net_if_get_config(iface)->ip.ipv4...